src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -345,7 +345,8 @@ private ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e)
                     HashSet<DiagnosticInfo>? useSiteDiagnostics = null;
                     if (expression.ConstantValue == ConstantValue.Null)
                     {
-                        if (inputType.IsNonNullableValueType())
+                        // Function pointers are value types, but they can be assigned null, so they can be matched against null.
+                        if (inputType.IsNonNullableValueType() && inputType.TypeKind != TypeKind.FunctionPointer)
```

Comment:
> if (inputType.IsNonNullableValueType() && inputType.TypeKind != TypeKind.FunctionPointer) [](start = 23, length = 90)

Are we not going through this code path for non-generic case?

Reply:
Nope. The first `if` in the function checks for type parameters, and goes down a different path for them.

---
In reply to: [541360178](https://github.com/dotnet/roslyn/pull/49915#discussion_r541360178) [](ancestors = 541360178)